deployer/src/persistence/mod.rs

```diff
@@ -248,10 +248,14 @@ impl Persistence {
         offset: u32,
         limit: u32,
     ) -> Result<Vec<Deployment>> {
-        let mut query = QueryBuilder::new("SELECT * FROM deployments WHERE service_id = ");
+        let mut query =
+            QueryBuilder::new("SELECT *, (SELECT COUNT(*) FROM deployments WHERE service_id = ");
 
+        let service_id = service_id.to_string();
         query
-            .push_bind(service_id.to_string())
+            .push_bind(&service_id)
+            .push(") as cnt FROM deployments WHERE service_id = ")
+            .push_bind(&service_id)
             .push(" ORDER BY last_update DESC LIMIT ")
             .push_bind(limit);
```

Comment:
LIMIT can be behind an if-case if the option is not unwrap_or'ed.

Reply:
> LIMIT can be behind an if-case if the option is not unwrap_or'ed.

Left this the same as it was not something I wrote, but what would we use for offset then? Because offset is calculated as `page * limit` so that logic would have to be changed, or is having no offset if limit is 0 fine?

Reply:
Leaving it is fine. I didn't see that there would be more changes involved.